File: leetcode/easy/2427. Number of Common Factors.ts
function commonFactors(a: number, b: number): number {
    if (a === 1 && b === 1) return 1;
    let count: number = 1;
    const max_num: number = Math.max(a,b);
    const min_num: number = Math.min(a,b);
    if (max_num % min_num === 0) count += 1;

    for (let div = 2; div <= Math.floor(min_num/2); div += 1){
        if (max_num % div === 0 && min_num % div === 0) count += 1;
    }
    return count;
};

File: leetcode/easy/2427. Number of Common Factors.rs
impl Solution {
    pub fn common_factors(a: i32, b: i32) -> i32 {
        if a == 1 && b == 1{
            return 1;
        }
        let mut smaller_num: i32 = a;
        let mut count: i32 = 1;
        if a > b{
            smaller_num = b;
            if a % b == 0{
                count += 1;
            }
        }else{
            if b % a == 0{
                count += 1;
            }
        }

        for div in 2..smaller_num/2+1{
            if a % div == 0 && b % div == 0{
                count += 1;
            }
        }
        return count;
    }
}
